src/net/sourceforge/kolmafia/maximizer/Evaluator.java

```diff
@@ -1411,7 +1398,8 @@ void enumerateEquipment(EquipScope equipScope, int maxPrice, PriceLevel priceLev
 
         boolean wrongClass = false;
         String classType = mods.getString(StringModifier.CLASS);
-        if (classType != "" && !classType.equals(KoLCharacter.getAscensionClassName())) {
+        if (!Objects.equals(classType, "")
```

Comment:
presumably `!classType.isEmpty()` is better to be more inline with the others, but why is IntelliJ recommending `Objects.equals` over `==` anyway?

Reply:
I redid. I wonder if it is because it thought mods,getString could return null?

Reply:
I wonder if it was done becausemods.getString could be null?

Reply:
That would usually be the reason